search/lib/convert/datetime.js

```diff
@@ -25,30 +25,53 @@ function splitOnDelimiters (str, delimiters) {
   return splits;
 }
 
+/**
+ * Handle single date case. End on the same day even if given a time.
+ */
+function handleSingleDate (dateTimes) {
+  const beginDate = new Date(dateTimes[0]);
+
+  const begin = beginDate.toISOString().replace('.000', '');
+
+  const endDate = new Date(beginDate);
+  endDate.setDate(endDate.getDate() + 1);
+  endDate.setUTCHours(0);
+  endDate.setMinutes(0);
+  endDate.setSeconds(0);
+  endDate.setMilliseconds(0);
+
+  const end = endDate.toISOString().replace('.000', '');
+
+  return { begin, end };
+}
+
+/**
+ * Handle a given range, only convert to valid CMR query string.
+ */
+function handleDateRange (dateTimes) {
+  const beginDate = new Date(dateTimes[0]);
+  const begin = beginDate.toISOString().replace('.000', '');
+
+  const endDate = new Date(dateTimes[1]);
+  const end = endDate.toISOString().replace('.000', '');
+
+  return { begin, end };
+}
+
 /**
  * Convert a datetime to a temporal range value.
  * If a single datetime is given, duplicate it to be the end of the range.
  * @param dateTimes
  * @returns {*}
  */
 function parseDateTimeHelper (dateTimes) {
-  let beginDate, endDate, begin, end;
+  let range;
   if (dateTimes.length === 1) {
-    beginDate = new Date(dateTimes[0]);
-
-    begin = beginDate.toISOString().replace('.000', '');
-
-    endDate = new Date(beginDate);
-    endDate.setDate(endDate.getDate() + 1);
-    end = endDate.toISOString().replace('.000', '');
+    range = handleSingleDate(dateTimes);
   } else {
-    beginDate = new Date(dateTimes[0]);
-    begin = beginDate.toISOString().replace('.000', '');
-
-    endDate = new Date(dateTimes[1]);
-    end = endDate.toISOString().replace('.000', '');
+    range = handleDateRange(dateTimes);
```

Comment:
So we only take the first 2 when given more than 1 datetime, approved as long as this is the correct behavior.